trollflow2/tests/test_trollflow2.py

```diff
@@ -2200,18 +2189,68 @@ def test_valid_filter(caplog, sc_3a_3b):
         assert "removing NIR016 for area euron1" in caplog.text
         assert "keeping IR037 for area euron1" in caplog.text
         assert "product absent not found, already removed" in caplog.text
-        tpg.reset_mock()
+
+
+def test_valid_filter_small_coverage(caplog, sc_3a_3b):
+    """Test filter for minimum fraction of valid data with small coverage."""
+    from trollflow2.plugins import check_valid_data_fraction
+
+    job, prods = _create_valid_filter_job_and_prods(sc_3a_3b)
+
+    with mock.patch("trollflow2.plugins._get_scene_coverage") as tpg, \
+            caplog.at_level(logging.DEBUG):
         tpg.return_value = 1
-        check_valid_data_fraction(job2)
+        check_valid_data_fraction(job)
         assert "inaccurate coverage estimate suspected!" in caplog.text
-        assert "NIR016" in prods2
-        assert "IR037" in prods2
-        tpg.reset_mock()
+        assert "NIR016" in prods
+        assert "IR037" in prods
+
+
+def test_valid_filter_zero_coverage(caplog, sc_3a_3b):
+    """Test filter for minimum fraction of valid data without any coverage."""
+    from trollflow2.plugins import check_valid_data_fraction
+
+    job, prods = _create_valid_filter_job_and_prods(sc_3a_3b)
+
+    with mock.patch("trollflow2.plugins._get_scene_coverage") as tpg, \
+            caplog.at_level(logging.DEBUG):
         tpg.return_value = 0
-        check_valid_data_fraction(job2)
+        check_valid_data_fraction(job)
         assert "no expected coverage at all, removing" in caplog.text
-        assert "NIR016" not in prods2
-        assert "IR037" not in prods2
+        assert "NIR016" not in prods
+        assert "IR037" not in prods
+
+
+def test_valid_filter_no_trollsched(caplog, monkeypatch, sc_3a_3b):
+    """Test filter for minimum fraction of valid data with full coverage."""
+    monkeypatch.setattr("trollsched.spherical.get_twilight_poly", None)
+    from trollflow2.plugins import check_valid_data_fraction
+
+    job, prods = _create_valid_filter_job_and_prods(sc_3a_3b)
+
+    with mock.patch("trollflow2.plugins._get_scene_coverage") as tpg, \
+            caplog.at_level(logging.DEBUG):
+        tpg.return_value = 100
+        check_valid_data_fraction(job)
+
+    assert "Trollsched import failed" in caplog.text
+    assert "Keeping all products" in caplog.text
```

Comment:
This seems to work only if it is run by it self. If all the tests are run, the tests fails.

Reply:
Fixed in https://github.com/pytroll/trollflow2/pull/223/commits/4058756893421c247ad399895a6cf23cc9a7ecd6